### DISCUSSION TITLE
embedding Javascript in a Wiki page

### DISCUSSION THREAD

**Discussion by borepstein** [requarks/wiki#6472] [Help / Questions] [answered]
Hello all,

I have some JS code (image gallery browser sort of thing in JQuery). The reference ot JQuery I put in the header in the theme code, the rest - in the box for the bottom of the body section. The errors do pop up in the console, the code itself would not start. I.e., the JQ .ready function would not start. Any thoughts?

Thanks.

Boris.

**Comment by borepstein** [comment]
I have solved it. The code was running but    I used the .ready   function and the did not work. The "on load"  event was the way to go.

```
$(window).on("load", function(){
  // Handler when all assets (including images) are loaded
});
```

Source: https://www.sitepoint.com/jquery-document-ready-plain-javascript/

**Reply by NGPixel** [reply, marked as answer]
That's not the proper way to do it and will likely fail for some users.

https://docs.requarks.io/en/troubleshooting#how-to-execute-javascript-code-on-page-load

**Reply by guoxiao0521** [reply]
In which file is this code written or how is it configured?
```javascript
window.boot.register('page-ready', () => {
    // code to execute
})
```

**Reply by NGPixel** [reply]
Admin > Theme

**Reply by cycne** [reply]
I’m sharing my implementation of an include feature for Wiki.js:

https://gist.github.com/cycne/5897c2fc39f27dbaecf7ee686e66913e
